curp/src/client_new/tests.rs

```diff
@@ -0,0 +1,644 @@
+        });
+    });
+    let unary = Unary::<TestCommand>::new(connects, None, None, None);
+    assert!(unary.local_connect().await.is_none());
+    let res = unary.fetch_cluster(false).await.unwrap();
+    assert_eq!(
+        res.into_members_addrs(),
+        HashMap::from([
+            (0, vec!["A0".to_owned()]),
+            (1, vec!["A1".to_owned()]),
+            (2, vec!["A2".to_owned()])
+        ])
+    );
+}
+
+#[traced_test]
+#[tokio::test]
+async fn test_unary_fetch_clusters_serializable_local_first() {
+    let connects = init_mocked_connects(3, |id, conn| {
+        conn.expect_fetch_cluster()
+            .return_once(move |_req, _timeout| {
+                let members = if id == 1 {
+                    // local server(1) does not see the cluster members
+                    vec![]
+                } else {
+                    panic!("other server's `fetch_cluster` should not be invoked");
+                };
+                Ok(tonic::Response::new(FetchClusterResponse {
+                    leader_id: Some(0),
+                    term: 1,
+                    cluster_id: 123,
+                    members,
+                    cluster_version: 1,
+                }))
+            });
+    });
+    let unary = Unary::<TestCommand>::new(connects, Some(1), None, None);
+    assert!(unary.local_connect().await.is_some());
+    let res = unary.fetch_cluster(false).await.unwrap();
+    assert!(res.members.is_empty());
+}
+
+#[traced_test]
+#[tokio::test]
+async fn test_unary_fetch_clusters_linearizable() {
+    let connects = init_mocked_connects(5, |id, conn| {
+        conn.expect_fetch_cluster()
+            .return_once(move |_req, _timeout| {
+                let resp = match id {
+                    0 => FetchClusterResponse {
+                        leader_id: Some(0),
+                        term: 2,
+                        cluster_id: 123,
+                        members: vec![
+                            Member::new(0, "S0", vec!["A0".to_owned()], false),
+                            Member::new(1, "S1", vec!["A1".to_owned()], false),
+                            Member::new(2, "S2", vec!["A2".to_owned()], false),
+                            Member::new(3, "S3", vec!["A3".to_owned()], false),
+                            Member::new(4, "S4", vec!["A4".to_owned()], false),
+                        ],
+                        cluster_version: 1,
+                    },
+                    1 | 4 => FetchClusterResponse {
+                        leader_id: Some(0),
+                        term: 2,
+                        cluster_id: 123,
+                        members: vec![], // linearizable read from follower returns empty members
+                        cluster_version: 1,
+                    },
+                    2 => FetchClusterResponse {
+                        leader_id: None, // imagine this node is a disconnected candidate
+                        term: 23,        // with a high term
```

Comment:
Xline has implemented the pre-vote phase in release 0.6.0, therefore it's impossible to happen.